include/xmipp4/core/service_catalog.hpp

```diff
@@ -49,26 +49,26 @@ class interface_catalog
      * a given type, it constructs a new concrete interface manager of that
      * type. If called previously, it returns the same instance.
      * 
-     * @tparam T interface manager. Must be child class of backend_manager.
+     * @tparam T interface manager. Must be child class of service_manager.
      * @return T& the requested interface manager.
      */
     template <typename T>
-    typename std::enable_if<std::is_convertible<T*, backend_manager*>::value, T&>::type
-    get_backend_manager();
+    typename std::enable_if<std::is_convertible<T*, service_manager*>::value, T&>::type
+    get_service_manager();
 
 private:
     class implementation;
     memory::pimpl<implementation> m_implementation;
```

Comment:
Replace `memory::pimpl` with std::unique_ptr. In the long term I'd like to deprecate `memory::pimpl`.